backend/src/middleware/validation.ts

```diff
@@ -30,11 +41,51 @@ const validateRequest: RequestHandler = (req, res, next) => {
   throw createHttpError(400, errorString);
 };
 
+const validateUser: RequestHandler = asyncHandler(async (req, res, next) => {
+  const authHeader = req.headers.authorization;
+  let token;
+
+  if (authHeader && authHeader.split(" ")[0] === "Bearer") {
+    token = authHeader.split(" ")[1];
+  } else {
+    throw createHttpError(401, "Authorization Token Required");
+  }
+
+  try {
+    const decodedToken = await firebaseAdmin.firebaseAuth.verifyIdToken(token);
+    const email = decodedToken.email;
+
+    const user = await UserModel.findOne({ email });
+    /* eslint-disable-next-line @typescript-eslint/no-unsafe-member-access */
+    req.body.currentUser = user;
+  } catch (error) {
+    throw createHttpError(404, "Error finding user");
+  }
+
+  next();
+});
+
+const validateHousingLocator: RequestHandler = (req, res, next) => {
+  validateUser(req, res, () => {
+    /* eslint-disable-next-line @typescript-eslint/no-unsafe-member-access */
+    if (req.body.currentUser.isHousingLocator) {
+      next();
+    } else {
+      res.status(401).send({ message: "Housing Locators only " });
+    }
+  });
+};
+
```

Comment:
Suggestion: What are your thoughts on moving these to a new file (`middleware/auth.ts`) and also renaming them to `requireUser` and `requireHousingLocator` or something similar just to make it clearer that these middleware aren't related to request content validation?